models.py

```diff
@@ -79,29 +79,36 @@ def cnn_model_fn(features, labels, mode, params):
     training = mode == tf.estimator.ModeKeys.TRAIN
     # loop for adding a convolutional layer and max pooling layer pair
     for layer in params["layers"]:
+        for conv in range(layer["num_convs"]):
```

Comment:
Two small comments, one very pedantic and one somewhat more interesting.

(i) Could you use single quotes / apostrophes when accessing dictionary fields (e.g. `params['layers']` and `layer['num_convs']`?  It doesn't matter to a huge degree, but the consistency with how I've been doing it suits my aesthetic sensibility :-).

(ii) I would also write the nested loop as `for _ in range(layer['num_convs'])`.  Currently, you're assigning to `conv` the integer values in the `range`; but notice that you never use that variable `conv`.  Using the underscore `_` allows you to run the loop without assigning any variables that then don't get used.